src/shared-testing/src/FakeApiClient.ts

```diff
@@ -45,6 +58,12 @@ export default class FakeApiClient implements ApiClient {
     this.errorFunctions = functionNames || []
   }
 
+  shouldReturnErrorAfterNCalls(error: Error, calls: number, functionNames?: string[]): void {
```

Comment:
This function name is a little misleading because it looks like from the fn signature then fn is returning void but the name suggests it should return an error?